src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -646,22 +672,20 @@ public ObjectId getLastCommitHash(VcsRepositoryUrl repoUrl) throws EntityNotFoun
         if (repoUrl == null || repoUrl.getURL() == null) {
             return null;
         }
-        // Get refs of repo without cloning it locally
-        Collection<Ref> refs;
+        // Get HEAD ref of repo without cloning it locally
         try {
             log.debug("getLastCommitHash " + repoUrl);
-            refs = Git.lsRemoteRepository().setRemote(getGitUriAsString(repoUrl)).setTransportConfigCallback(sshCallback).call();
+            var headRef = Git.lsRemoteRepository().setRemote(getGitUriAsString(repoUrl)).setTransportConfigCallback(sshCallback).callAsMap().get("HEAD");
```

Comment:
is there no method in JGit to retrieve the head more easily?

Reply:
See comment below; I'm unsure why you need the repository to retrieve it's default branch :)

Reply:
Not every repository has the default branch which is set for the VCS, this is sadly not guaranteed